### PR TITLE
libcgroup: update to 2.0.

### PR DESCRIPTION
**Pull request by wibed**
update libcgroup to version 2.0

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [x] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/libcgroup/patches/musl-decls.patch [hunk 1]
```diff
@@ -18,5 +18,5 @@ include file sys/cdefs.h
 +#endif /* !defined(__GLIBC__) */
 +
  #define _LIBCGROUP_H_INSIDE
- 
+
```
Comment (paper42) [inline srcpkgs/libcgroup/patches/musl-decls.patch:21]: why?

File: srcpkgs/libcgroup/patches/musl-sterror_r.patch [hunk 2]
```diff
@@ -4,7 +4,7 @@ GNU libc's if _GNU_SOURCE is defined.
 --- a/src/api.c	2014-01-13 15:05:56.000000000 +0100
 +++ b/src/api.c	2015-10-25 14:09:18.690232596 +0100
 @@ -3332,8 +3332,16 @@
- 
+
```
Comment (paper42) [inline srcpkgs/libcgroup/patches/musl-sterror_r.patch:7]: why?

File: srcpkgs/libcgroup/template [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
 hostmakedepends="flex automake libtool"
 makedepends="pam-devel"
 short_desc="Library that abstracts the control group file system in Linux"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="LGPL-2.1"
-homepage="http://libcg.sourceforge.net"
-distfiles="${SOURCEFORGE_SITE}/libcg/${pkgname}-${version}.tar.bz2"
-checksum=e4e38bdc7ef70645ce33740ddcca051248d56b53283c0dc6d404e17706f6fb51
-
+homepage="https://github.com/${pkgname}/${pkgname}"
```
Comment (paper42) [inline srcpkgs/libcgroup/template:11]: don't use $pkgname in homepage

File: srcpkgs/libcgroup/template [hunk 4]
```diff
@@ -1,18 +1,16 @@
...
+distfiles="${homepage}/releases/download/v2.0/${pkgname}-${version}.tar.bz2"
```
Comment (paper42) [inline srcpkgs/libcgroup/template:12]: I wouldn't use $homepage and $pkgname. 2.0 should be replaced with $version. tar.gz is prefered over bz2.

**Comment by paper42** [comment]
pre_configure is not necessary, because you use a tarball with pregenerated configure script. The commit message body is not necessary.

File: srcpkgs/libcgroup/patches/musl-decls.patch (hunk 1 above)
Comment (wibed) [reply srcpkgs/libcgroup/patches/musl-decls.patch]: cp paste creep

File: srcpkgs/libcgroup/patches/musl-sterror_r.patch (hunk 2 above)
Comment (wibed) [reply srcpkgs/libcgroup/patches/musl-sterror_r.patch]: cp paste creep

File: srcpkgs/libcgroup/template (hunk 3 above)
Comment (wibed) [reply srcpkgs/libcgroup/template]: pgkname removed from var

File: srcpkgs/libcgroup/template (hunk 4 above)
Comment (wibed) [reply srcpkgs/libcgroup/template]: i kept $homepage and $pkgname because it shortens the url quite a bit

Comment (paper42) [reply srcpkgs/libcgroup/template]: I don't think we have a shortage of bytes.

Comment (wibed) [reply srcpkgs/libcgroup/template]: it would be nice to keep the 80 column mark as close as possible.

File: srcpkgs/libcgroup/patches/musl-sterror_r.patch [hunk 5]
```diff
@@ -17,6 +16,6 @@ GNU libc's if _GNU_SOURCE is defined.
 +		return "strerror_r() failed";
 +	}
 +#endif
- 
+
```
Comment (paper42) [inline srcpkgs/libcgroup/patches/musl-sterror_r.patch:19]: please revert the changes on patches

File: srcpkgs/libcgroup/template (hunk 4 above)
Comment (paper42) [reply srcpkgs/libcgroup/template]: distfiles often go over that mark and that's fine.

File: srcpkgs/libcgroup/template [hunk 6]
```diff
@@ -1,18 +1,16 @@
...
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="LGPL-2.1"
-homepage="http://libcg.sourceforge.net"
-distfiles="${SOURCEFORGE_SITE}/libcg/${pkgname}-${version}.tar.bz2"
-checksum=e4e38bdc7ef70645ce33740ddcca051248d56b53283c0dc6d404e17706f6fb51
-
+homepage="https://github.com/libcgroup/libcgroup"
+distfiles="${homepage}/releases/download/v${version}/${pkgname}-${version}.tar.gz"
+checksum=aecc501a9ea6a97da0673585db5081df912ae607dec36d5f6f7ab14f69d48ab8
 case "$XBPS_TARGET_MACHINE" in
```
Comment (paper42) [inline srcpkgs/libcgroup/template:14]: add an empty line between checksum= and case.

File: srcpkgs/libcgroup/patches/musl-sterror_r.patch (hunk 5 above)
Comment (wibed) [reply srcpkgs/libcgroup/patches/musl-sterror_r.patch:20]: the file has been reverted as in the original

File: srcpkgs/libcgroup/template (hunk 4 above)
Comment (wibed) [reply srcpkgs/libcgroup/template]: distfile now contains full url

File: srcpkgs/libcgroup/template (hunk 6 above)
Comment (wibed) [reply srcpkgs/libcgroup/template]: added

**Comment by wibed** [comment]
> pre_configure is not necessary, because you use a tarball with pregenerated configure script. The commit message body is not necessary.

do you mean the block? 
```
pre_configure() {
  autoreconf -if
}
```

**Comment by paper42** [comment]
> > pre_configure is not necessary, because you use a tarball with pregenerated configure script. The commit message body is not necessary.
> 
> do you mean the block?
> 
> ```
> pre_configure() {
>   autoreconf -if
> }
> ```

yes and you might be able to remove the hostmakedepends variable then

**Comment by wibed** [comment]
> > > pre_configure is not necessary, because you use a tarball with pregenerated configure script. The commit message body is not necessary.
> > 
> > 
> > do you mean the block?
> > ```
> > pre_configure() {
> >   autoreconf -if
> > }
> > ```
> 
> yes and you might be able to remove the hostmakedepends variable then

without the pre_configure block it builds fine
but it depends on the hostmakedepends variable

**Comment by paper42** [comment]
> > > pre_configure is not necessary, because you use a tarball with pregenerated configure script. The commit message body is not necessary.
> > 
> > yes and you might be able to remove the hostmakedepends variable then
> 
> without the pre_configure block it builds fine
> but it depends on the hostmakedepends variable

sorry, I forgot about flex, we can drop automake and libtool, but have to leave flex there

```
hostmakedepends="flex"
```

Can you also fix the lint issue from CI?

> srcpkgs/libcgroup/template:10: use SPDX id for 'LGPL-2.1' license or see Manual.md

File: srcpkgs/libcgroup/patches/musl-sterror_r.patch (hunk 5 above)
Comment (paper42) [reply srcpkgs/libcgroup/patches/musl-sterror_r.patch:20]: Something is wrong, GitHub still shows the change. Are you sure you reverted the changes?

**Comment by paper42** [comment]
@wibed are you still interested in continuing to work on this PR?

**Comment by wibed** [comment]
> @wibed are you still interested in continuing to work on this PR?

@paper42 i do, please be patient.

**Comment by wibed** [comment]
@mt3 you just comitted literally when i was to to push my changes upstream

**Comment by wibed** [comment]
@mt3 it seems the checksum should be updated

File: srcpkgs/libcgroup/template (hunk 6 above)
Comment (paper42) [reply srcpkgs/libcgroup/template:14]: this was not solved

**Review by paper42 — COMMENTED** [review]
please squash the commits to one and rebase on top of master, so CI can run

File: srcpkgs/libcgroup/template (hunk 6 above)
Comment (paper42) [reply srcpkgs/libcgroup/template:14]: this was not solved

sorry for the double comment

**Comment by paper42** [comment]
> @mt3 it seems the checksum should be updated

yes, that should fix the CI

File: srcpkgs/libcgroup/template [hunk 7]
```diff
@@ -1,17 +1,16 @@
...
 makedepends="pam-devel"
 short_desc="Library that abstracts the control group file system in Linux"
 maintainer="Orphaned <orphan@voidlinux.org>"
-license="LGPL-2.1"
-homepage="http://libcg.sourceforge.net"
-distfiles="${SOURCEFORGE_SITE}/libcg/${pkgname}-${version}.tar.bz2"
-checksum=e4e38bdc7ef70645ce33740ddcca051248d56b53283c0dc6d404e17706f6fb51
+license="LGPL-2.1-only"
+homepage="https://github.com/libcgroup/libcgroup"
+distfiles="https://github.com/libcgroup/libcgroup/releases/download/v${version}/libcgroup-${version}.tar.bz2"
```
Comment (paper42) [inline srcpkgs/libcgroup/template:12]: let's use tar.gz

**Review by paper42 — COMMENTED** [review]
squash the commits and it should be good to go

**Comment by wibed** [comment]
okey, despise the "no push" commit message. i have reverted the commit back to 2a257f4, because i have committed and pushed from root directory. after that i have rebased to master and pushed up again.

**Comment by wibed** [comment]
it is getting late

**Comment by ericonr** [comment]
How are you trying to rebase and squash the commits? We have a lot of comments in PRs here explaining how to do it correctly, if you need the exact commands.

**Comment by wibed** [comment]
@ericonr 

i usually do a : `git rebase -i HEAD~N` .

but i get all the other commits, because i am not in my own branch.
overreaching my pr.

is there another way i am not aware of?

**Comment by wibed** [comment]
ill reopen a new pr
